aten/src/ATen/native/cuda/ForeachReduceOp.cu

```diff
@@ -44,6 +45,191 @@ struct TensorListAddresses {
   const void* addresses[MAX_TENSORS_PER_KERNEL];
 };
 
+template <
+    typename T,
+    int depth = 1,
+    int r_args_depth = 1,
+    int res_arg_index = 0>
+struct LpMaxFunctor {
+  __device__ __forceinline__ void operator()(
+      int chunk_size,
+      TensorListMetadata<depth>& tl,
+      T* output_per_tensor_ptr,
+      const int max_chunks_per_tensor) {
+    const auto tensor_loc = tl.block_to_tensor[blockIdx.x];
+    const auto chunk_idx = tl.block_to_chunk[blockIdx.x];
+    auto n = tl.numel_for_tensor[tensor_loc];
+
+    T* x = (T*)tl.addresses[0][tensor_loc];
+    x += chunk_idx * chunk_size;
+    n -= chunk_idx * chunk_size;
+
+    __shared__ T s_vals[512];
+    T vals[kILP];
+    T r_x[kILP];
+    for (int64_t i = 0; i < kILP; i++) {
+      vals[i] = T(-INFINITY);
+      r_x[i] = T(-INFINITY);
```

Comment:
could be replaced with `std::numeric_limits<T>::min()`

Reply:
experimented and used `lowest` in https://github.com/pytorch/pytorch/pull/127303